src/components/Series/BarSeries.tsx

```diff
@@ -60,18 +60,20 @@ export function BarSeries(props: BarSeriesProps) {
   }, [color, displayMarkers, figure]);
   const isVisible = useIsSeriesVisible(id);
   useEffect(() => {
-    legendDispatch({
-      type: 'ADD_LEGEND_LABEL',
-      payload: {
-        id,
-        label: otherProps.label,
-        colorLine,
-        shape,
-      },
-    });
+    if (!hidden) {
+      legendDispatch({
+        type: 'ADD_LEGEND_LABEL',
+        payload: {
+          id,
+          label: otherProps.label,
+          colorLine,
+          shape,
+        },
+      });
+    }
     return () =>
       legendDispatch({ type: 'REMOVE_LEGEND_LABEL', payload: { id } });
```

Comment:
the cleanup callback should also be in the `if` block (same in other files)

Reply:
thank you, the bug is fixed now